### DISCUSSION TITLE
Benutzerdefiniertes Werkzeug mit parametrisiertem DataLinq Aufruf aus WebGIS möglich?

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#153] [Q&A]
Wir haben versucht, ein benutzerdefiniertes Werkzeug für die Grundstückseigentümer-Abfrage über DataLinq zu realisieren.
Die Idee ist, DataLinq einen Aufrufparameter mitzugeben, welcher die KG-Nummer und Grundstücksnummer enthält. Laut [Doku](https://docs.webgiscloud.com/cloud/webgis_apps/mapviewer/customjs/customtools.html#werkzeuge-mit-karteninteraktion) kann man nur Parameter wie {X} {Y} usw. mitgeben. Wir würden uns aber wünschen, dass man beim Klick in die Karte auch ein im CMS vordefiniertes Suchfeld eines gView Dienstes als Parameter übergeben kann. Bei Grundstücken zB die Kombination aus KG-Nummer und Grundstücksnummer ["kg_gst"]. Also eigentlich sehr ähnlich wie die Aktion die passiert, wenn man das Abfragen – Identifizieren Werkzeug verwendet.
So etwa in dieser Form: datalinq/report/eigentümerabfrage@eigentümer@view?kgngstnr={kg_gst}

Müssen wir da selbst etwas bauen oder gibt es eine Möglichkeit Attribute analog der Variante {X}, {Y} bei den benutzerdefinierten Werkzeugen zu übergeben?

**Comment by jugstalt** [comment]
Hallo Jürgen,

das funktioniert so nicht. Ein benutzerdefiniertes Werkzeug hat eigentlich keinen Zugriff auf die Abfragen. Es müsste ja hier erst einmal eine Abfrage auf ein Grundstück ausgeführt werden und dann deine DataLinq Seite aufgerufen werden. Aber was passiert, wenn mehrere Objekte durch den Klick (Klicktoleranz) betroffen sind? Bei GrundstückensFLÄCHEN vielleicht nicht so ein Problem, aber wenn man das Tool schon vereinheitlicht, sollte man das schon bedenken...

Also die Antwort ist:
Nein benutzerdefinierte Werkzeuge haben keinen Zugriff auf die Daten! Das bleibt auch so.

Workaround:
* Du übergibst die Koordinaten an eine "Zwischenseite" wo du dich selber darum kümmerst, dass damit die Grundstücke abgeholt werden und leitest dann auf die DataLinq Seite mit den entsprechenden Attribute weiter.

* Empfohlener Weg: Kein benuterdefiniertes Werkzeug sondern Link aus der Attributtabelle. Genauso fragen wir die Eigentümer intern auch ab.  Anwender selektiert im WebGIS Grundstücke und springt von dort ins DataLinq oder die Online GDB Abfrage etc. Das ist durchgängiger. Anwender muss keine neuen Werkzeuge "lernen", sondern springt immer über Tabelle zum weiterführenden Seiten. Diese können dann auch einfach in einem Popup angezeigt werden und er muss den Viewer nicht jedes mal verlassen. Es macht meiner Meinung sowieso wenig Sinn für jedes Funktion immer gleich ein benutzerdefiniertes Werkzeug anzubieten.
